src/ziggurat/streams.clj

```diff
@@ -235,11 +241,12 @@
   (let [topology-fn (case (:consumer-type stream-config)
                       :stream-joins stream-joins-topology
                       topology)
-        top         (topology-fn handler-fn stream-config topic-entity channels)
-        _           (log/info (.describe top))]
-    (KafkaStreams. ^Topology top
-                   ^Properties (properties stream-config)
-                   (new TracingKafkaClientSupplier tracer))))
+        top         (topology-fn handler-fn stream-config topic-entity channels)]
+
+    (when-not (nil? top)
```

Comment:
what happens when it is nil?

Reply:
when it returns nil we will not create the topology, it will throw an error if we try to cast it to a Topology object